### PR TITLE
travis: update distro for new clang-format

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

**Comment by junghans** [comment]
As excepted still clang-7 even Ubuntu 20.04, let install the package explicitly.

**Comment by junghans** [comment]
format this please

**Comment by junghans** [comment]
@dalg24 @rfbird now everything uses clang-10 for formatting.

File: CMakeLists.txt [hunk 1]
```diff
@@ -69,7 +69,7 @@ Cabana_add_dependency( PACKAGE MPI )
 Cabana_add_dependency( PACKAGE ArborX )
 
 # find Clang Format
-Cabana_add_dependency( PACKAGE CLANG_FORMAT )
+find_package( CLANG_FORMAT )
```
Comment (sslattery) [inline CMakeLists.txt:72]: Does this imply that this is now a requirement?

File: .travis.yml [hunk 2]
```diff
@@ -67,7 +67,8 @@ addons:
       - libopenmpi-dev
       - libgsl0-dev
       - libfftw3-dev
-      - python-pip
+      - python3-pip
+      - clang-format
```
Comment (sslattery) [inline .travis.yml:71]: Does this need to specifically be `clang-format-10`?

**Review by sslattery — COMMENTED** [review]
A few questions.

File: CMakeLists.txt (hunk 1 above)
Comment (junghans) [reply CMakeLists.txt:72]: No. The clang_format just doesn't need our magic to create an `ENABLE_*` options, plus one could even see anymore what version was used, but we could certainly undo this change.

File: .travis.yml (hunk 2 above)
Comment (junghans) [reply .travis.yml:71]: We could do that, usually the clang never gets bump within a ubuntu release, so that should be ok for a while.

**Review by sslattery — APPROVED** [review]

**Comment by rfbird** [comment]
I think we still need to specify pointer alignment, as right not it is not prescribed (and may be set to derived)

**Pull request merged**